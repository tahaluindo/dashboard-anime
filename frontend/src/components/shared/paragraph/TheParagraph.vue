<template>
  <section
    class="grid items-center justify-center text-current grid-cols-1 gap-10 lg:grid-cols-3 font-opensans clamptext-085-12 leading-loose"
  >
    <div v-if="hasFirst" :class="{ ...grid, major: major === 'First' }">
      <slot name="first" />
    </div>
    <div v-if="hasSecond" :class="{ ...grid, major: major === 'Second' }">
      <slot name="second" />
    </div>
  </section>
</template>

<script lang="ts">
import { computed, defineComponent, PropType } from "vue";

export default defineComponent({
  props: {
    hasFirst: {
      type: Boolean,
      default: true,
    },
    hasSecond: {
      type: Boolean,
      default: true,
    },
    major: {
      type: String as PropType<"First" | "Second">,
      default: "Second",
      validator: (value: string) => ["First", "Second"].indexOf(value) !== -1,
    },
  },
  setup(props) {
    const grid = computed(() => {
      return {
        "col-span-full":
          (props.hasFirst && !props.hasSecond) ||
          (!props.hasFirst && props.hasSecond),
      };
    });

    return {
      grid,
    };
  },
});
</script>

<style lang="postcss">
.major {
  @apply col-span-2;
}
</style>
