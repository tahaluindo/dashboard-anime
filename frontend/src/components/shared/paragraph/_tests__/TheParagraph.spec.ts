import { mount } from "@vue/test-utils";
import TheParagraph from "../TheParagraph.vue";

describe("TheParagraph test suit", () => {
  it("should render the Paragraph if has first and second parameters", () => {
    const wrapper = mount(TheParagraph, {
      slots: {
        first: "<div> FIRST </div>",
        second: "<div> SECOND </div>",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render only the first element if hasSecond equals to false", () => {
    const wrapper = mount(TheParagraph, {
      slots: {
        first: "<div> FIRST </div>",
        second: "<div> SECOND </div>",
      },
      props: {
        hasSecond: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render only the second element if hasFirst equals to false", () => {
    const wrapper = mount(TheParagraph, {
      slots: {
        first: "<div> FIRST </div>",
        second: "<div> SECOND </div>",
      },
      props: {
        hasFirst: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render first element with major class if the major setted to First", () => {
    const wrapper = mount(TheParagraph, {
      slots: {
        first: "<div> FIRST </div>",
        second: "<div> SECOND </div>",
      },
      props: {
        major: "First",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
