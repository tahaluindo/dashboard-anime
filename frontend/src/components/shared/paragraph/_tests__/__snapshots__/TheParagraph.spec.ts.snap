// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheParagraph test suit should render first element with major class if the major setted to First 1`] = `
"<section class=\\"grid items-center justify-center text-current grid-cols-1 gap-10 lg:grid-cols-3 font-opensans clamptext-085-12 leading-loose\\">
  <div class=\\"major\\">
    <div> FIRST </div>
  </div>
  <div class=\\"\\">
    <div> SECOND </div>
  </div>
</section>"
`;

exports[`TheParagraph test suit should render only the first element if hasSecond equals to false 1`] = `
"<section class=\\"grid items-center justify-center text-current grid-cols-1 gap-10 lg:grid-cols-3 font-opensans clamptext-085-12 leading-loose\\">
  <div class=\\"col-span-full\\">
    <div> FIRST </div>
  </div>
  <!--v-if-->
</section>"
`;

exports[`TheParagraph test suit should render only the second element if hasFirst equals to false 1`] = `
"<section class=\\"grid items-center justify-center text-current grid-cols-1 gap-10 lg:grid-cols-3 font-opensans clamptext-085-12 leading-loose\\">
  <!--v-if-->
  <div class=\\"col-span-full major\\">
    <div> SECOND </div>
  </div>
</section>"
`;

exports[`TheParagraph test suit should render the Paragraph if has first and second parameters 1`] = `
"<section class=\\"grid items-center justify-center text-current grid-cols-1 gap-10 lg:grid-cols-3 font-opensans clamptext-085-12 leading-loose\\">
  <div class=\\"\\">
    <div> FIRST </div>
  </div>
  <div class=\\"major\\">
    <div> SECOND </div>
  </div>
</section>"
`;
